File: src/main/java/com/oracle/ocmjea/gof/behavioral/command/FileSystemReceiverUtil.java
package com.oracle.ocmjea.gof.behavioral.command;

public class FileSystemReceiverUtil {

	public static FileSystemReceiver getSistemaOperacionalDaMaquina() {
		String osName = System.getProperty("os.name");
		System.out.println("Sistema Operacional: " + osName);
		
		if (osName.contains("Windows")) {
			return new WindowsFileSystemReceiver();
		} else {
			return new UnixFileSystemReceiver();
		}
	}
}

File: design-patterns/src/main/kotlin/designpatterns/behavioral/command/FileSystemReceiverUtil.kt
package designpatterns.behavioral.command

object FileSystemReceiverUtil {
    fun getOperatingSystem(): FileSystemReceiver {
        val osName = System.getProperty("os.name")
        println("Operating System: $osName")
        return if (osName.contains("Windows")) {
            WindowsFileSystemReceiver()
        } else {
            UnixFileSystemReceiver()
        }
    }
}
